### DISCUSSION TITLE
Versioning Selector with Chromatic

### DISCUSSION THREAD

**Discussion by yhy-1** [storybookjs/storybook#25222] [Help] [answered]
### Summary

Hi, I have been looking around for documentation on how to have a versioning selector using composition but the documentation isn't helpful. It said "Change the version of the composed Storybook to see how the library evolves. This requires [configuration](https://storybook.js.org/docs/sharing/package-composition#show-a-version-selector) from the package author." The links just link back to each other. Does anyone have implemented this and can provide an example of how it done?

As shown in the image below:  
![Snag_6454b71](https://github.com/storybookjs/storybook/assets/52711139/834643f5-80bf-47ee-a69c-99c7a2f07940)

**Comment by chantastic** [comment, marked as answer]
“Composition” is a feature of Storybook that lets you compose other Storybooks into your Storybook Sidebar.
This is often used in a case where you’re building a Storybook but using a library like MUI (which has its own Storybook).
In that situation, you can lock to a specific version of that document.

More info here: https://storybook.js.org/docs/sharing/storybook-composition

**Reply by yhy-1** [reply]
I have been looking at that document but I can't find the part about the versioning. Is it locked only at that specific version? I want to let the user be able to select between versions. I.E. similar to the photo. The part I find is here (https://storybook.js.org/docs/sharing/package-composition#show-a-version-selector) but they just link back to each other and said I need to set some configuration. I can't find what configuration I need to set to make that happen.

@chantastic

**Reply by chantastic** [reply]
I'm trying to say that this isn't a part of Storybook unless you are using Storybook Composition to import a a third-party Storybook. Is that what you are attempting?

**Reply by yhy-1** [reply]
@chantastic  Currently my storybook has one version which is the latest in the main branch. Using chromatic, I want to be able to deploy older versions of my storybook. (https://storybook.js.org/docs/sharing/publish-storybook#versioning-and-history). I would expect to be able to show those versions using package-compostion. Which I thought was the case since the image above was taken from storybook documentation from  https://storybook.js.org/docs/sharing/package-composition. 

Doesn't look like it possible?
Sample repo: https://github.com/yhy-1/storybook-versioning

**Reply by chantastic** [reply]
no. this is not the case. this is what i’m trying to tell you. you would
need to do versioned deploys thru your hosting/CI process.

even Storybook Composition is not a version selector. it is simply an
indicator of the third party storybook you are locked to.

i want to be very clear. the feature you are reading about in documentation
is not what you expect it to be. if you want to provide versioned
documentation, you need to build that functionality or use a hosting
service like Chromatic that does branch deploys by default.
